datasources/yellowstone-grpc-datasource/src/lib.rs

```diff
@@ -128,6 +128,10 @@ impl Datasource for YellowstoneGrpcGeyserClient {
                         match result {
                             Ok((mut subscribe_tx, mut stream)) => {
                                 while let Some(message) = stream.next().await {
+                                    if cancellation_token.is_cancelled() {
```

Comment:
because this is ongoing while loop, the code on line `123` never gets checked. We need to break from the loop to make sure that the datasource is cancelled:

```rust
  _ = cancellation_token.cancelled() => {
      log::info!("Cancelling Yellowstone gRPC subscription.");
      break;
  }
```